packages/admin-panel/src/editor/Editor.js

```diff
@@ -35,9 +35,9 @@ export const Editor = ({ fields, recordData, onEditField }) => {
     <div>
       {fields
         .filter(({ show = true, editConfig = {} }) => {
-          const { visibilityCriteria } = editConfig;
+          const { visibilityCriteria, hideForEdit } = editConfig;
 
-          if (!show) {
+          if (!show || hideForEdit) {
```

Comment:
Added condition for a new property in 'editConfig' object. As 'show' property applies across the whole field object, 'hideForEdit' can only apply for the edit modal. Thus we can show the 'child_id' in the admin panel table, but not in the edit modal.